Library/ViewModels/PledgeAmountViewModel.swift

```diff
@@ -107,25 +109,30 @@ public final class PledgeAmountViewModel: PledgeAmountViewModelType,
       .map(third)
       .skipRepeats()
 
-    self.doneButtonIsEnabled = updatedValue
-      .map { min, max, doubleValue in min <= doubleValue && doubleValue <= max }
+    let isValueValid = updatedValue
+      .map { (min: Double, max: Double, doubleValue: Double) -> Bool in
+        min <= doubleValue && doubleValue <= max
+      }
 
-    self.stepperStepValue = self.stepperMinValue
+    self.doneButtonIsEnabled = isValueValid
 
-    let clampedStepperValue = Signal.combineLatest(
-      self.stepperMinValue,
-      self.stepperMaxValue,
-      self.textFieldValueProperty.signal.skipNil()
-    )
-    .map(clampedValue)
+    let textColor = isValueValid
+      .map { $0 ? UIColor.ksr_green_500 : UIColor.ksr_red_400 }
+
+    self.labelTextColor = textColor
+
+    self.stepperStepValue = minValue
 
     self.stepperValue = Signal.merge(
-      initialValue,
-      clampedStepperValue
+      minValue,
+      textFieldValue
     )
 
     self.textFieldIsFirstResponder = self.doneButtonTappedProperty.signal
       .mapConst(false)
+
+    self.textFieldTextColor = textColor
+      .wrapInOptional()
```

Comment:
Unfortunately it seemed necessary to pass `textColor` as an optional to a `UITextField` (which is how it's also defined in UIKit)